lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/EvaluatorRequest.java

```diff
@@ -146,11 +146,23 @@ public String getRuntimeName() {
     private final List<String> nodeNames = new ArrayList<>();
     private final List<String> rackNames = new ArrayList<>();
     private String runtimeName = "";
+    private EvaluatorRequestor requestor = null;
 
     private Builder() {
     }
 
     /**
+     * Sets its requestor before setting parameters and building.
+     * Useful for using fluid interface.
+     *
+     * @param requestor the EvaluatorRequestor
+     * @return this Builder
+     */
+    public Builder(final EvaluatorRequestor requestor) {
```

Comment:
Please mark this and the other new method in this class with `@Private`. They are not part of the `public` API of REEF, for which we promise compatibility between any two subsequent versions.